### DISCUSSION TITLE
The bandwidth of KVStore

### DISCUSSION THREAD

**Discussion by shadow3x3x3** [nats-io/nats.go#1478] [Q&A]
Hi,

I am develping an app base on natsKV. BTW, it's awsome. 

Is it possible to modify the bandwidth usage of the natKV client? For instance, by utilizing `nats.RateLimit`.

I've observed that the `SubOpts` used in the current kv `Subscribe` cannot be externally adjusted. Is it possible to offer a way for external control over `Subscribe` options within the kv? 

Thanks!

**Comment by ripienaar** [comment]
The rate limit option is only for push consumers, we're moving away from those to pull requests in the new APIs and those do not have the Rate limit concept

**Reply by shadow3x3x3** [reply]
So, natsKV can't control the rate in client side(golang)? How about server side?

**Reply by ripienaar** [reply]
Not server side, I guess you want to limit watchers?

**Reply by shadow3x3x3** [reply]
Yeah. Can we restrict the watch rate of the watcher?

**Reply by ripienaar** [reply]
In old API no, in new API I suspect its managed due to it being pull based but we should ask @piotrpio if there's a way influence fetch interval
